x-pack/plugins/security_solution/server/usage/collector.ts

```diff
@@ -226,43 +226,172 @@ export const registerCollector: RegisterCollector = ({
+              },
+              exponential_average_bucket_processing_time_ms: {
+                type: 'long',
+                _meta: {
+                  description:
+                    'Exponential moving average of all bucket processing times, in milliseconds',
+                },
+              },
+              exponential_average_bucket_processing_time_per_hour_ms: {
+                type: 'long',
+                _meta: {
+                  description:
+                    'Exponentially-weighted moving average of bucket processing times calculated in a 1 hour time window, in milliseconds',
+                },
+              },
+              maximum_bucket_processing_time_ms: {
+                type: 'long',
+                _meta: {
+                  description: 'Maximum among all bucket processing times, in milliseconds',
+                },
+              },
+              minimum_bucket_processing_time_ms: {
+                type: 'long',
+                _meta: {
+                  description: 'Minimum among all bucket processing times, in milliseconds',
+                },
+              },
+              total_bucket_processing_time_ms: {
+                type: 'long',
+                _meta: { description: 'Sum of all bucket processing times, in milliseconds' },
+              },
             },
             datafeed: {
-              datafeed_id: { type: 'keyword' },
-              state: { type: 'keyword' },
+              datafeed_id: {
+                type: 'keyword',
+                _meta: {
+                  description: 'A numerical character string that uniquely identifies the datafeed',
+                },
+              },
+              state: {
+                type: 'keyword',
+                _meta: { description: 'The status of the datafeed' },
+              },
               timing_stats: {
-                average_search_time_per_bucket_ms: { type: 'long' },
-                bucket_count: { type: 'long' },
-                exponential_average_search_time_per_hour_ms: { type: 'long' },
-                search_count: { type: 'long' },
-                total_search_time_ms: { type: 'long' },
+                average_search_time_per_bucket_ms: {
+                  type: 'long',
+                  _meta: { description: 'The average search time per bucket, in milliseconds' },
+                },
+                bucket_count: {
+                  type: 'long',
+                  _meta: { description: 'The number of buckets processed' },
+                },
+                exponential_average_search_time_per_hour_ms: {
+                  type: 'long',
+                  _meta: {
+                    description: 'The exponential average search time per hour, in milliseconds',
+                  },
+                },
+                search_count: {
+                  type: 'long',
+                  _meta: { description: 'The number of searches run by the datafeed' },
```

Comment:
NIT: Should we mention when the counter is reset?

Reply:
I don't think it should reset. It is unique to the Datafeed. But I could be mistaken

https://www.elastic.co/guide/en/elasticsearch/reference/current/ml-get-datafeed-stats.html#ml-get-datafeed-stats-results